src/libraries/Directory.Build.props

```diff
@@ -355,6 +355,7 @@
   <PropertyGroup Condition="'$(BuildAllConfigurations)' != 'true'">
     <!-- We add extra binplacing for the test shared framework until we can get hardlinking with the runtime directory working on all platforms -->
     <BinPlaceTestSharedFramework Condition="'$(BuildTargetFramework)' == '$(NetCoreAppCurrent)'">true</BinPlaceTestSharedFramework>
+    <BinPlaceTestSharedFramework Condition="'$(BuildAllProjects)' != 'true' and '$(TargetFramework)' != '$(NetCoreAppCurrent)'"></BinPlaceTestSharedFramework>
```

Comment:
What are you doing here?  Binlpacing should be smart enough to understand if the TargetFramework you are building is the best one for NetCoreAppCurrent.  If it's getting overridden when TargetFramework is specified then we should fix the BinPlaceConfiguration.